cfn-templates/management-account-iam.yaml

```diff
@@ -0,0 +1,181 @@
+---
+AWSTemplateFormatVersion: "2010-09-09"
+
+Description: >-
+ Provisions IAM resources within the management account to assume the necessary
+ roles within accounts in a development environment. This provisions IAM policies
+ and groups that users can be placed in.
+
+ Because this does not directly create any users or roles, it does not need to be
+ aware of any Permissions Boundaries.
+
+Parameters:
+  ContractorGroupPrefix:
+    Description: >-
+      The prefix to apply to the IAM user group that will contain contractor users.
+    Type: String
+  GovernmentGroupPrefix:
+    Description: >-
+      The prefix to apply to the IAM user group that will contain Government users.
+    Type: String
+
+Resources:
+  IamSelfAdminPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      Description: Allow users to modify their own IAM configuration
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Sid: AllowViewAccountInfo
+            Effect: Allow
+            Action:
+              - iam:GetAccountPasswordPolicy
+              - iam:ListVirtualMFADevices
+            Resource: "*"
```

Comment:
Sonar was grumpy about this. I marked this as safe with the following justification:

> Both of the actions in this statement are required in order for users to be able modify their own IAM permissions. Neither of these actions support filtering based on resources (the account password policy doesn't have an ARN and the other call is a List action) according the Service Authorization Reference (https://docs.aws.amazon.com/service-authorization/latest/reference/list_identityandaccessmanagement.html).